src/helpers/buffers/unordered_receiver.rs

```diff
@@ -0,0 +1,502 @@
+use crate::{
+    helpers::{messaging::Message, Error},
+    protocol::RecordId,
+};
+use futures::{task::Waker, Future, Stream};
+use generic_array::GenericArray;
+use pin_project::pin_project;
+use std::{
+    marker::PhantomData,
+    num::NonZeroUsize,
+    pin::Pin,
+    sync::{Arc, Mutex},
+    task::{Context, Poll},
+};
+use typenum::Unsigned;
+
+/// A future for receiving item `i` from an `UnorderedReceiver`.
+#[pin_project]
```

Comment:
It feels that you don't need to pin anything. `usize` is `Unpin` and `OperatingState` pins the stream

Reply:
Hmm, I added this early on and didn't come back to it (what you see is not what I started with).  I'll take another look.